src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure implements GetArea, Draw {
+
+    private double firstLeg;
+    private double secondLeg;
+    private double height;
+
+    public double getFirstLeg() {
+        return firstLeg;
+    }
+
+    public void setFirstLeg(double firstLeg) {
+        if (firstLeg > 0) {

```

The setter methods for firstLeg, secondLeg, and height do not handle the case where the input is less than or equal to zero. It silently ignores invalid values. Consider throwing an `IllegalArgumentException` if the input is invalid to ensure the object is always in a valid state.